profiler/options.go

```diff
@@ -146,6 +147,7 @@ func logStartup(c *config) {
 		"execution_trace_size_limit": c.traceConfig.Limit,
 		"endpoint_count_enabled":     c.endpointCountEnabled,
 		"custom_profiler_label_keys": c.customProfilerLabels,
+		"enable":                     c.enable,
```

Comment:
The same change needs to be added to `profiler/telemetry.go`.

Reply:
Not sure I follow. Having `enable` in the debug log is useful for debugging, e.g. when a customer reports that profiling isn't working.

And having it in telemetry is useful for us to understand how our users use this flag?

Reply:
I also think keeping this information in the debug log would make it easier to handle future reports about the profiler.
I'm fine with either option, so please feel free to share your thoughts and I'll implement them accordingly!

Reply:
> Having enable in the debug log is useful for debugging, e.g. when a customer reports that profiling isn't working.

Agreed. I added this comment too hastily and didn't notice that the startup log happens before the check for `DD_PROFILING_ENABLED`. Let's keep this after all.

> And having it in telemetry is useful for us to understand how our users use this flag?

As this PR stands right now, the telemetry client won't start if `DD_PROFILING_ENABLED=false`. My gut feeling is that we shouldn't start telemetry if we don't start the profiler. IMO this is lower priority than making sure `DD_PROFILING_ENABLED=false` works, and we can address telemetry in a followup. WDYT @felixge?

Reply:
SGTM.

Reply:
I've also added it to `profiler/telemetry.go`.
[`3aac285` (#2840)](https://github.com/DataDog/dd-trace-go/pull/2840/commits/3aac285f204da839371965213b30a1401345f45b)